### PR TITLE
Remove ClusterDeployment controller

### PR DESCRIPTION
**Pull request by ninabauer**

### CODE REVIEW

File: pkg/util/mocks/structs/builders.go [hunk 1]
```diff
@@ -22,9 +22,8 @@ const (
 	testProject = "project"
 	testRegion  = "us-east1"
 
-	clusterPlatformLabel          = "hive.openshift.io/cluster-platform"
-	clusterPlatformGCP            = "gcp"
-	clusterDeploymentManagedLabel = "api.openshift.com/managed"
+	clusterPlatformLabel = "hive.openshift.io/cluster-platform"
+	clusterPlatformGCP   = "gcp"
```
Comment (rafael-azevedo) [inline pkg/util/mocks/structs/builders.go:26]: These two variables can be removed.

File: pkg/util/util.go [hunk 2]
```diff
@@ -38,25 +38,6 @@ func GetSecret(kubeClient client.Client, secretName, namespace string) (*corev1.
...
-			Namespace: namespace,
-		},
-		Data: map[string][]byte{
-			"osServiceAccount.json": []byte(creds),
-		},
-	}
-}
-
 // NewGCPSecretCRV2 returns a Secret CR formatted for GCP for use in projectreference controller.
 func NewGCPSecretCRV2(creds string, namespacedNamed kubetypes.NamespacedName) *corev1.Secret {
```
Comment (rafael-azevedo) [inline pkg/util/util.go:42]: Can we rename this function to  NewGCPSecretCR and adjust the comments as well

File: pkg/util/mocks/structs/builders.go (hunk 1 above)
Comment (rafael-azevedo) [reply pkg/util/mocks/structs/builders.go:26]: Actually I think this whole file can be removed

File: pkg/util/util.go (hunk 2 above)
Comment (ninabauer) [reply pkg/util/util.go]: Renamed function and comment

File: pkg/util/mocks/structs/builders.go (hunk 1 above)
Comment (ninabauer) [reply pkg/util/mocks/structs/builders.go]: Removed the variables.

**Pull request merged**